components/prompt_based_laion_retrieval/src/main.py

```diff
@@ -81,18 +45,40 @@ def transform(
             modality=Modality.IMAGE,
         )
 
-        meta_df = pd.DataFrame(columns=["id", "url"], dtype="string")
-        meta_df = meta_df.set_index("id")
-
-        dataframe = dataframe["prompts_text"].map_partitions(
-            query_clip_client,
-            client=client,
-            meta=meta_df
-        )
-
-        dataframe = dataframe.rename(columns={"url": "images_url"})
+    def transform(
+            self,
+            dataframe: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """
+        Args:
+            dataframe: Dask dataframe
 
-        return dataframe
+        Returns:
+            Dask dataframe
+        """
+        results: t.List[t.Tuple[str]] = []
+        loop = asyncio.new_event_loop()
+
+        async def async_query():
```

Comment:
async is cool but hopefully not required since it is not easy to write

Reply:
This was already in the implementation of the component, so no change here.